internal/pkg/cli/app_deploy.go

```diff
@@ -341,16 +341,24 @@ func (o *appDeployOpts) pushToECRRepo() error {
 }
 
 func (o *appDeployOpts) getAppDockerfilePath() (string, error) {
+	type dfPath interface {
+		DockerfilePath() string
+	}
+
 	manifestBytes, err := o.workspaceService.ReadAppManifest(o.AppName)
 	if err != nil {
 		return "", fmt.Errorf("read manifest file %s: %w", o.AppName, err)
 	}
 
-	mf, err := manifest.UnmarshalApp(manifestBytes)
+	app, err := manifest.UnmarshalApp(manifestBytes)
 	if err != nil {
 		return "", fmt.Errorf("unmarshal app manifest: %w", err)
 	}
 
+	mf, ok := app.(dfPath)
```

Comment:
Why do we need this validation? Will we have a type of application without dockerfile path?

Reply:
Ideally we shouldn't, but if we call `app.DockerfilePath()` and the new app struct (let's say `BackendApp`) doesn't have the method provided then the CLI will panic.

Instead, by checking if the interface is satisfied we will get a more explicit error returned to the user so it will make debugging easier.

This is a good question though, how can we guarantee this method is always present on all structs from the `manifest` pkg?
One thing I can do is add a test for `manifest.UnmarshalApp` for each app type to make sure the interface is implemented.

Reply:
maybe instead of having the `unmarshalApp` to return a generic interface, we can let it return

```
type app interface {
   DockerfilePath()
}
```

Reply:
It's just that there are other methods that people might need when `UnmarshalApp` happens like `ApplyEnv` :| hence the returned empty interface